### PR TITLE
Adds redis lock

### PR DESCRIPTION
**Pull request by codingjoe**
Adds a new lock based on redis' lock capabilities.

### CODE REVIEW

File: viewflow/lock.py [hunk 1]
```diff
@@ -77,3 +95,29 @@ def lock(flow_cls, process_pk):
...
+
+
+def redis_lock(flow, attempts=5, expires=120):
+    """Task lock based redis locks implemented in ``django-redis``."""
+    @contextmanager
+    def lock(flow_cls, process_pk):
+        key = 'django-viewflow-lock-{}/{}'.format(flow_cls._meta.namespace, process_pk)
+
+        for i in range(attempts):
+            lock = cache.lock(key, timeout=expires)
```
Comment (samifahed) [inline viewflow/lock.py]: @codingjoe what if `cache.lock` is not there?

Comment (samifahed) [inline viewflow/lock.py]: they have to be using `django-redis` only right?

Comment (codingjoe) [inline viewflow/lock.py]: yes, I a wrote in the docstring. I could thorw a custom exception.

Comment (samifahed) [inline viewflow/lock.py]: I will merge it as is now, since we are using it this way, but eventually we will need to throw an exception

**Pull request closed without merging**